animation_nodes/data_structures/meshes/mesh_data.pyx

```diff
@@ -132,35 +135,53 @@ cdef class Mesh:
         else:
             raise Exception("invalid length")
 
-    def insertUVMap(self, str name, Vector2DList uvs):
-        if len(uvs) == len(self.polygons.indices):
-            self.uvMaps[name] = uvs
+    def insertAttribute(self, str name, str type, str domain, str dataType, object data):
```

Comment:
Use a guard condition, and use enums instead of str.

```python
if referenceAmount != len(data):
    raise Exception("invalid length")

attribute = Attribute()
...
...
```

Reply:
Done.